packages/hooks/useTimeout.js

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react';
+
+export default (ms = 0) => {
+  const [ready, setReady] = useState(false);
+
+  useEffect(
+    () => {
+      const timer = setTimeout(() => {
```

Comment:
I would add a `boolean` here  called `unmounting` or something and set it to `false`. Once the timeout is up make sure to check that `unmounting` isn't set to true before you call the state update or else you will run into warnings.